tfhe/src/high_level_api/tests/noise_distribution/encryption_noise.rs

```diff
@@ -0,0 +1,232 @@
+use crate::shortint::parameters::test_params::{
+    TEST_PARAM_KEYSWITCH_PKE_TO_BIG_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M128,
+    TEST_PARAM_KEYSWITCH_PKE_TO_SMALL_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M128,
+    TEST_PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M128,
+    TEST_PARAM_PKE_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M128,
+};
+use crate::shortint::parameters::{
+    AtomicPatternParameters, ClassicPBSParameters, CompactPublicKeyEncryptionParameters,
+    ShortintKeySwitchingParameters,
+};
+use crate::{ClientKey, CompactCiphertextList, CompactPublicKey, ConfigBuilder, FheUint2};
+
+use rayon::prelude::*;
+
+#[test]
+fn test_noise_check_secret_key_encryption_noise_tuniform() {
+    let params: AtomicPatternParameters = TEST_PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M128.into();
+
+    let modulus_as_f64 = if params.ciphertext_modulus().is_native_modulus() {
+        2.0f64.powi(u64::BITS as i32)
+    } else {
+        params.ciphertext_modulus().get_custom_modulus() as f64
+    };
+
+    let encryption_noise = params.encryption_noise_distribution();
+
+    let (inclusive_min_val, inclusive_max_val, expected_variance) = match encryption_noise {
+        DynamicDistribution::Gaussian(_gaussian) => {
+            panic!("This test is written for TUniform, wrong parameter set used")
+        }
+        DynamicDistribution::TUniform(tuniform) => (
+            tuniform.min_value_inclusive(),
+            tuniform.max_value_inclusive(),
+            tuniform.variance(modulus_as_f64),
+        ),
+    };
+
+    let config = ConfigBuilder::with_custom_parameters(params).build();
+
+    const NB_TEST: usize = 125_000;
+
+    let noise_samples: Vec<_> = (0..NB_TEST)
+        .into_par_iter()
+        .map(|_| {
+            let cks = ClientKey::generate(config);
+            let encrypted = FheUint2::encrypt(0u32, &cks);
+
+            // Drop to lower level APIs to get to the noise
+            let integer_key = cks.into_raw_parts().0;
+            let shortint_key: &crate::shortint::ClientKey = integer_key.as_ref();
+
+            let radix = encrypted.into_raw_parts().0;
+            assert!(radix.blocks.len() == 1);
+            let shortint_block = radix.blocks.into_iter().next().unwrap();
+
+            let noise = shortint_key.decrypt_no_decode(&shortint_block);
+            let signed_noise: i64 = noise.0 as i64;
+
+            assert!(
+                signed_noise >= inclusive_min_val,
+                "{signed_noise} is not >= {inclusive_min_val}"
+            );
+
+            assert!(
+                signed_noise <= inclusive_max_val,
+                "{signed_noise} is not <= {inclusive_max_val}"
+            );
+
+            // Rescale
+            signed_noise as f64 / modulus_as_f64
+        })
+        .collect();
+
+    let measured_variance = variance(&noise_samples);
+    // With our test settings this gives a range of around 1/100 of a bit, this should be accurate
```

Comment:
Not sure what you mean by "1/100 of a bit"

Is it that `max_bound.sqrt() - min_bound.sqrt() ≈ encoding_step / 100`?